src/test/java/uk/ac/ebi/eva/pipeline/io/writers/AnnotationMongoWriterTest.java

```diff
@@ -188,17 +188,27 @@ public void shouldWriteAllFieldsIntoMongoDbMultipleSetsAnnotations() throws Exce
     public void shouldWriteSubstitutionScoresIntoMongoDb() throws Exception {
         String databaseName = mongoRule.getRandomTemporaryDatabaseName();
 
-        Annotation annotation = new Annotation("X", 1, 10, "A", "T",
-                VEP_VERSION, VEP_CACHE_VERSION);
+        ScoreMongo siftScore = new ScoreMongo(0.02, "deleterious");
+        ScoreMongo polyphenScore = new ScoreMongo(0.846, "possibly_damaging");
 
-        Score siftScore = new Score(0.02, "deleterious");
-        Score polyphenScore = new Score(0.846, "possibly_damaging");
-
-        ConsequenceType consequenceType = new ConsequenceType();
+        ConsequenceTypeMongo consequenceType = new ConsequenceTypeMongo(null, null,
+            null, null, null, null, null, null,
+            null, null, null, null, null, null);
         consequenceType.setSift(siftScore);
         consequenceType.setPolyphen(polyphenScore);
 
-        annotation.addConsequenceType(consequenceType);
+        AnnotationMongo annotation = new AnnotationMongo(
+            AnnotationMongo.buildAnnotationId("X", 1, "A", "T",
+                VEP_VERSION, VEP_CACHE_VERSION),
+            "X",
+            1,
+            10,
+            VEP_VERSION,
+            VEP_CACHE_VERSION,
+            null,
+            Collections.singleton(consequenceType));
+
+        annotation = annotation.concatenate(annotation);
```

Comment:
Again I am a bit unclear on why we append an Annotation object to itself...

Reply:
Concatenate method calls within it `addConsequenceTypes` which calls `addConsequenceType` for each consequenceType in the consequenceType set. `AddConsequenceType` also sets the Xref fields for AnnotationMongo object, besides just copying over ConsequenceType information. So that's why we call the `concatenate` function, and not return the annotation object directly (so as to set Xref fields as well).